packages/nimble-components/src/table-column/number-text/index.ts

```diff
@@ -180,6 +179,7 @@ export class TableColumnNumberText extends mixinTextBase(
                 alignment: this.determineCellContentAlignment(),
                 placeholder: this.placeholder
             };
+            this.columnInternals.headerAlignment = columnConfig.alignment;
             this.columnInternals.columnConfig = columnConfig;
```

Comment:
Curious why we should have both a number column specific `columnConfig.alignment` property and a shared across all columns `columnInternals.headerAlignment` property. Seems like the number-text column should just use the columnInternals value and the columnInternals value should be more generic (`columnInternals.alignment`).

Thoughts @m-akinc @mollykreis on if there is a reason to keep them separate?

Reply:
The cell view doesn't have access to the full `ColumnInternals` object -- just the `TableColumnNumberTextColumnConfig`. I think that's the main reason. The table's template passes the alignment value from `ColumnInternals` to the header, and it can't access a property that is only in certain column types' config object.

Reply:
This enterprising individual thought otherwise:

https://github.com/ni/nimble/blob/f47da47d1ee6104e9e4c913c8e1fbca321825e9d/packages/nimble-components/src/table-column/base/cell-view/index.ts#L59

But it does make more sense for the cellView instances to not have direct coupling to column / columnInternals. I'll create a tech debt to break that coupling that was introduced with delegated events. So the approach in this PR seems good.

Reply:
😅

Reply:
Created tech debt https://github.com/ni/nimble/issues/2258